GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Branches.cs

```diff
@@ -53,6 +38,44 @@ protected BaseBranchNode(Tree tree, string fullPath)
                 var dirs = fullPath.Split(PathSeparator);
                 Name = dirs[dirs.Length - 1];
                 ParentPath = dirs.Take(dirs.Length - 1).Join(PathSeparator.ToString());
+                Visible = visible;
+            }
+
+            protected string AheadBehind { get; set; }
+
+            /// <summary>
+            /// Short name of the branch/branch path. <example>"issue1344"</example>.
+            /// </summary>
+            public string Name { get; protected set; }
+
+            protected string ParentPath { get; }
+
+            /// <summary>
+            /// Full path of the branch. <example>"issues/issue1344"</example>.
+            /// </summary>
+            public string FullPath => ParentPath.Combine(PathSeparator.ToString(), Name);
+
+            /// <summary>
+            /// Gets whether the commit that the node represents is currently visible in the revision grid.
+            /// </summary>
+            public bool Visible { get; }
+
+            protected override void ApplyStyle()
```

Comment:
Just curious: Do we sort members alphabetically regardless of their visibility now?

Reply:
"Now" as in - in the official public release - or as in - in the new implementation?

Sorting was introduced in https://github.com/gitextensions/gitextensions/pull/8427, and this is available in the Dev stream. The official version has only one sorting mode - A-Z.

Reply:
> "Now" as in - in the official public release - or as in - in the new implementation?

Oh, this was confusing in two ways!
I meant "newly", i.e. the latter.
But the question was about the coding style whether to sort class members ignoring `private`, `protected`, ...

Reply:
The general rule I follow (almost OCD about this) is A-Z order within groups as per the original [MS Design Guidelines](https://docs.microsoft.com/en-us/archive/blogs/brada/internal-coding-guidelines#28-file-organization), i.e.: 
```cs
public class MyClass : IFoo
{
      // fields
      int foo;

      // constructors
      public MyClass() { … }

      // properties
      public int Bar { get { … } set { … } }
      public int Foo { get { … } set { … } }
      public int Rex { get { … } set { … } }

      // events
      public event EventHandler FooChanged { add { … } remove { … } }

      // methods
      public void DoSomething() { … }
      public void FindSomethind() { … }

      protected void FindSomethindProtected() { … }

      private void DoSomethingPrivate() { … }

      //private interface implementations
      void IFoo.DoSomething() { DoSomething(); }

      // nested types
      class NestedType { … }
}
```

But our codebase has very diverse set of ordering practices, and even our dotnet/* repos no longer follow a single ordering style (which drives me crazy, hehe).

Reply:
At the same time, when making changes to the code I have to leave members in their place to provide a meaningful diff.
I will clean up the code once this work is complete.